contracts/mainnet/connectors/morpho-blue/main.sol

```diff
@@ -0,0 +1,918 @@
+	 * @param _setId ID stores the amount of tokens repaid.
+	 */
+	function repay(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt;
+		(
+			_marketParams, // Updated token contracts in case of Eth
+			_amt // Assets final amount to repay
+		) = _performEthToWethConversion(
+			_marketParams,
+			_assets,
+			address(this),
+			_getId,
+			Mode.Repay
+		);
+
+		// Approving loan token for repaying
+		approve(
+			TokenInterface(_marketParams.loanToken),
+			address(MORPHO_BLUE),
+			_amt
+		);
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.repay(
+			_marketParams,
+			_amt,
+			0,
+			address(this),
+			new bytes(0)
+		);
+
+		setUint(_setId, _assets);
+
+		_eventName = "LogRepay((address,address,address,address,unit256),uint256,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_assets,
+			_shares,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @notice Repays assets on behalf.
+	 * @dev The market to repay assets to. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _marketParams The market to repay assets to.
+	 * @param _assets The amount of assets to repay. (For max: `uint256(-1)`)
+	 * @param _onBehalf The address whose loan will be repaid.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens repaid.
+	 */
+	function repayOnBehalf(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		address _onBehalf,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt;
+		(
+			_marketParams, // Updated token contracts in case of Eth
+			_amt // Assets final amount to repay
+		) = _performEthToWethConversion(
+			_marketParams,
+			_assets,
+			_onBehalf,
+			_getId,
+			Mode.Repay
+		);
+
+		// Approving loan token for repaying
+		approve(
+			TokenInterface(_marketParams.loanToken),
+			address(MORPHO_BLUE),
+			_amt
+		);
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.repay(
+			_marketParams,
+			_amt,
+			0,
+			_onBehalf,
+			new bytes(0)
+		);
+
+		setUint(_setId, _assets);
+
+		_eventName = "LogRepayOnBehalf((address,address,address,address,unit256),uint256,uint256,address,uint256,uint256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_assets,
+			_shares,
+			_onBehalf,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @notice Repays shares on behalf.
+	 * @dev The market to repay assets to. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _marketParams The market to repay assets to.
+	 * @param _shares The amount of shares to burn. (For max: `uint256(-1)`)
+	 * @param _onBehalf The address whose loan will be repaid.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens repaid.
+	 */
+	function repayOnBehalfShares(
```

Comment:
there's no way to repay max similar to the withdraw max feature?

Reply:
The max repayment logics are being handled inside the `_performEthToWeth` conversion functions. It will check the minimum of balance and the debt amount and use that to repay. Logic can be found [here](https://github.com/Instadapp/dsa-connectors-2.0/pull/1/files#diff-57afa2d25dc241f0573736f488fd0f69847c6dfeeffa95efbdd7bc0fe9fec13eR59-R77).